online_store/gt_sport_v2.js

```diff
@@ -0,0 +1,139 @@
+const sportProducts = [
+  {
+    imgSrc: "skateboard.jpg",
+    name: "Skateboard",
+    brand: "Hudora",
+    level: "beginner",
+    price: 50,
+    seller: "Hudora",
+    other: {
+      helmetNeeded: true,
+      longevity: "mid-term",
+      teamSpirit: false,
+    }
+  },
+  {
+    imgSrc: "standuppaddle.jpg",
+    name: "Stand-up paddle",
+    brand: "Oceana",
+    level: "advanced",
+    price: 300,
+    seller: "Oceana",
+    other: {
+      helmetNeeded: false,
+      longevity: "long-term",
+      teamSpirit: true,
+    }
+  },
+  {
+    imgSrc: "volleyball.jpg",
+    name: "Volleyball ball",
+    brand: "Mikasa",
+    level: "beginner",
+    price: 26,
+    seller: "Mikasa",
+    other: {
+      helmetNeeded: false,
+      longevity: "short-term",
+      teamSpirit: true,
+    }
+  },
+  {
+    imgSrc: "xbadmintonset.jpg",
+    name: "Badminton set",
+    brand: "Talbot Torro",
+    level: "intermediate",
+    price: 28,
+    seller: "Talbot Torro",
+    other: {
+      helmetNeeded: false,
+      longevity: "mid-term",
+      teamSpirit: true,
+    }
+  },
+  {
+    imgSrc: "skigoggles.jpg",
+    name: "Ski Goggles",
+    brand: "Lemego",
+    level: "intermediate",
+    price: 28,
+    seller: "Lemego",
+    other: {
+      helmetNeeded: true,
+      longevity: "mid-term",
+      teamSpirit: true,
+    }
+  },
+];
+const productDescription = document.getElementById("productDescription");
+let shoppingCart = [];
+let productEl = [];
+const sportProductsEl = document.getElementById("sportProducts");
+const addToShoppingCartEl = document.getElementById("addToShoppingCart");
+let totalPriceEl = document.getElementById("totalPrice").value;
+let shoppingCartEl = document.getElementById("shoppingCart");
+let intoSCbtn = document.getElementById("intoSC");
+let calTPbtn = document.getElementById("calTP");
+let addToShoppingCartBtn = document.getElementById("addToShoppingCartBtn");
+let descriptions = [];
+
+function productDescriptionDisplay() {
+  for (let i = 0; i < sportProducts.length; i++) {
+    descriptions = document.createElement("div");
+    const btnEl = document.createElement("button");
+    const product = sportProducts[i];
+    descriptions.innerHTML = `
+      <p> <b>${sportProducts[i].name}</b></p>
+      <p> <img src='${sportProducts[i].imgSrc}' style="width:50px"><p>
+      <p> <b>Brand:</b> ${sportProducts[i].brand}</p>
+      <p> <b>Price:</b> ${sportProducts[i].price} Eur</p>
+      <p> Sold by a <b>gT partner</b> called ${sportProducts[i].seller}.</p>
+      <p> This ${sportProducts[i].name} is suitable for teenagers on ${sportProducts[i].level} level.
+      `;
+
+      const addToShoppingCartBtn = document.createElement("button");
+      addToShoppingCartBtn.textContent = "Add to shopping Cart";
+      addToShoppingCartBtn.addEventListener("click", addProductsToCart);
```

Comment:
Here we have a problem because we are passing the `addProductsToCart` function as a callback to the `addEventListener`. The `addEventListener` is expecting the callback to be a function that gets the DOMEvent. So we end up adding dom events to the cart not products 🙃. 

We could solve this in many ways but how about creating an anonymous function as the callback function and inside that new anonymous fn we call addProductsToCart with the product we already have in scope. I'm not adding the code because I want to challenge you. 💪

Reply:
Ok, would this work?
`    addToShoppingCartBtn.addEventListener('click', function () {
      shoppingCart.push(product);
};`

Reply:
solved :)